daemon/daemon_test.go

```diff
@@ -544,3 +602,91 @@ func (s *daemonSuite) TestGracefulStop(c *check.C) {
 		c.Fatal("never got proper response")
 	}
 }
+
+func (s *daemonSuite) TestRestartSystemWiring(c *check.C) {
+	d := newTestDaemon(c)
+	// mark as already seeded
+	s.markSeeded(d)
+
+	l, err := net.Listen("tcp", "127.0.0.1:0")
+	c.Assert(err, check.IsNil)
+
+	snapdAccept := make(chan struct{})
+	d.snapdListener = &witnessAcceptListener{Listener: l, accept: snapdAccept}
+
+	snapAccept := make(chan struct{})
+	d.snapListener = &witnessAcceptListener{Listener: l, accept: snapAccept}
+
+	d.Start()
+	defer d.Stop()
+
+	snapdDone := make(chan struct{})
+	go func() {
+		select {
+		case <-snapdAccept:
+		case <-time.After(2 * time.Second):
+			c.Fatal("snapd accept was not called")
+		}
+		close(snapdDone)
+	}()
+
+	snapDone := make(chan struct{})
+	go func() {
+		select {
+		case <-snapAccept:
+		case <-time.After(2 * time.Second):
+			c.Fatal("snap accept was not called")
+		}
+		close(snapDone)
+	}()
+
+	<-snapdDone
+	<-snapDone
+
+	oldRebootWaitTimeout := rebootWaitTimeout
+	defer func() {
+		execCommand = exec.Command
+		rebootWaitTimeout = oldRebootWaitTimeout
+	}()
+	rebootWaitTimeout = 100 * time.Millisecond
+
+	var cmds [][]string
+	execCommand = func(name string, arg ...string) *exec.Cmd {
```

Comment:
(nitpick) using testutil.MockCommand() might be slightly less code.

Reply:
done, in the end because of the change to the signature to reboot the easiest was to mock reboot() itself here, and have a separate test using MockCommand to test reboot() itself